frontend/model/contracts/chatroom.js

```diff
@@ -213,14 +213,20 @@ sbp('chelonia/defineContract', {
         if (!memberID) {
           throw new Error('The new member must be given either explicitly or implcitly with an inner signature')
         }
+
         if (!state.shouldSaveMessages) {
+          if (!state.members) {
+            Vue.set(state, 'members', {})
+          }
           if (state.members[memberID]) {
             throw new GIChatroomAlreadyMemberError(`Can not join the chatroom which ${memberID} is already part of`)
           }
         }
 
         Vue.set(state.members, memberID, { joinedDate: meta.createdDate })
 
+        if (!state.attributes) return
```

Comment:
This patch solves the issue of the action failing to process, but you may want to add a comment explaining why this is needed (and perhaps I'd also use `has(state, 'attributes')` instead).